libs/converter-nh-accuvote/src/convert/accuvote.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  assert,
+  err,
+  ok,
+  Optional,
+  Result,
+  resultBlock,
+} from '@votingworks/basics';
+import { safeParseNumber } from '@votingworks/types';
+import { decode as decodeHtmlEntities } from 'he';
+import { ConvertIssue, ConvertIssueKind } from './types';
+
+/**
+ * `AVSInterface` element data.
+ */
+export interface AvsInterface {
+  accuvoteHeaderInfo: AccuvoteHeaderInfo;
+  candidates: Candidates[];
+  ballotPaperInfo?: BallotPaperInfo;
+}
+
+/**
+ * `AccuvoteHeaderInfo` element data.
+ */
+export interface AccuvoteHeaderInfo {
+  electionDate: string;
+  electionName: string;
+  townName: string;
+  townId: string;
+  partyName?: string;
+  precinctId?: string;
+  electionId: string;
+  ballotType?: string;
+  ballotSize: string;
+}
+
+/**
+ * `Candidates` element data.
+ */
+export interface Candidates {
+  officeName: OfficeName;
+  candidateNames: CandidateName[];
+}
+
+/**
+ * `OfficeName` element data.
+ */
+export interface OfficeName {
+  name: string;
+  winnerNote?: string;
+  x: number;
+  y: number;
+}
+
+/**
+ * `CandidateName` element data.
+ */
+export interface CandidateName {
+  name: string;
+  pronunciation?: string;
+  party?: string;
+  writeIn?: boolean;
+  cx: number;
+  cy: number;
+  ox: number;
+  oy: number;
+}
+
+/**
+ * `BallotPaperInfo` element data.
+ */
+export interface BallotPaperInfo {
+  questions: string;
+}
+
+function getOptionalChildText(
+  element: Element,
+  tagName: string
+): Optional<string> {
+  for (const child of Array.from(element.childNodes)) {
+    if (child.nodeType === 1 && child.nodeName === tagName) {
+      const textContent = child.textContent?.trim();
+
+      if (!textContent || textContent === '') {
+        return undefined;
+      }
+
+      return textContent;
+    }
+  }
+}
+
+function getOptionalChild(
+  element: Element,
+  tagName: string
+): Optional<Element> {
+  for (const child of Array.from(element.childNodes)) {
+    if (child.nodeType === 1 && child.nodeName === tagName) {
+      return child as Element;
+    }
+  }
+}
```

Comment:
Minor nit: could use `getOptionalChild` in `getOptionalChildText`, since there's some repeated logic